lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -1,31 +1,37 @@
 class MiqAeDomain < MiqAeNamespace
   default_scope { where(:parent_id => nil).where(arel_table[:name].not_eq("$")) }
   validates_inclusion_of :parent_id, :in => [nil], :message => 'should be nil for Domain'
+  # TODO: Once all the specs start passing in the tenant object, enforce its presence
+  # validates_presence_of :tenant, :message => "object is needed to own the domain"
   after_destroy :squeeze_priorities
-  default_value_for(:priority) { MiqAeDomain.highest_priority + 1 }
   default_value_for :system,  false
   default_value_for :enabled, false
+  before_save :default_priority
   belongs_to :tenant
 
   def self.enabled
     where(:enabled => true)
   end
 
-  def self.reset_priority_by_ordered_ids(ids)
+  def self.reset_priority_by_ordered_ids(ids, tenant)
     ids.each_with_index do |id, priority|
-      MiqAeDomain.where(:id => id).first.try(:update_attributes, :priority => priority + 1)
+      MiqAeDomain.where(:id => id, :tenant => tenant).first.try(:update_attributes, :priority => priority + 1)
```

Comment:
@kbrock @gmcculloug 
Yes makes sense, since we have the sorted ids we don't need the tenant object and also the caller doesn't have to pass it in. This also changes the controller. I will change it